### PR TITLE
[Bug] [transform] Recover unequal code transform in FakeStream

### LINKED ISSUE #1140: [Bug] [plugin] recover unequal code transform in FakeStream
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

There is an unequal code transform in PR https://github.com/apache/incubator-seatunnel/pull/1041/files#diff-cc7eda78c56362a980afe456e101eaf8d5697aa16045a967ebe3b29bfea58760

### SeaTunnel Version

2.0.5

### SeaTunnel Config

```conf
N/A
```

### Running Command

```shell
N/A
```

### Error Exception

```log
N/A
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yx91490**
## Purpose of this pull request
close #1140 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by xleoken** [comment]
Hi, did you meet some exception? It seems not clear why we need recover this change.

**Comment by xleoken** [comment]
hi @simon824, can you help to review when you are free.

File: seatunnel-connectors/seatunnel-connector-spark-fake/src/main/scala/org/apache/seatunnel/spark/source/FakeStream.scala [hunk 1]
```diff
@@ -56,7 +56,11 @@ class FakeStream extends SparkStreamingSource[String] {
   }
 
   override def checkConfig(): CheckResult = {
-    checkAllExists(config, "content")
+    if (config.hasPath("content") && config.getStringList("content").nonEmpty) {
+      CheckResult.success()
```
Comment (simon824) [inline seatunnel-connectors/seatunnel-connector-spark-fake/src/main/scala/org/apache/seatunnel/spark/source/FakeStream.scala:60]: Maybe we can check if the type of `config.getAnyRef("content")`  is` List`  in `CheckConfigUtil.checkAllExists()`, which can be in common use.

Comment (yx91490) [reply seatunnel-connectors/seatunnel-connector-spark-fake/src/main/scala/org/apache/seatunnel/spark/source/FakeStream.scala:60]: Firstly let's make it correctly by recover it, then finish your idea in another issue may be better.

Comment (yx91490) [reply seatunnel-connectors/seatunnel-connector-spark-fake/src/main/scala/org/apache/seatunnel/spark/source/FakeStream.scala:60]: @simon824  what do you think ?

Comment (simon824) [reply seatunnel-connectors/seatunnel-connector-spark-fake/src/main/scala/org/apache/seatunnel/spark/source/FakeStream.scala:60]: Actually, this problem exist in many other connectors, it's unnecessary to fix all connectors `checkConfig` one by one.  
#1214 this pr has fix similar problem.

Comment (yx91490) [reply seatunnel-connectors/seatunnel-connector-spark-fake/src/main/scala/org/apache/seatunnel/spark/source/FakeStream.scala:60]: this pr seems not bad, it's suitable for this issue, but put all checks to CheckConfigUtil.checkAllExists() may cause lacking of type check, for example, if you want to get a intList, but provide stringList.
If you get a good idea about how to check all param's existance along with it's type, please tell me, thanks.

**Comment by yx91490** [comment]
this pr was broken, continue to https://github.com/apache/incubator-seatunnel/pull/1237

**Pull request closed without merging**